src/utils/config.js

```diff
@@ -0,0 +1,190 @@
+  // name (PascalCase) and prop name (camelCase)
+  BAlert: { variant: 'info' },
+  BBadge: { variant: 'secondary' },
+  BButton: { variant: 'secondary' },
+  BButtonClose: { textVariant: null, ariaLabel: 'Close' },
+  BDropdown: { variant: 'secondary' },
+  BFormFile: {
+    browseText: 'Browse',
+    dropPlaceholder: 'Drop files here',
+    placeholder: 'No file chosen' // Chrome default file prompt
+  },
+  BImg: { blankColor: 'transparent' },
+  BImgLazy: { blankColor: 'transparent' },
+  MBodal: {
+    cancelTitle: 'Cancel',
+    cancelVariant: 'secondary',
+    okTitle: 'OK',
+    okVariant: 'primary',
+    headerCloseLabel: 'Close'
+  }
+}
+
+// This contains user defined configuration
+//
+// Should this be stored here, or on the Vue.prototype ?????
+// For testing purposes, we might want to store this on Vue, so
+// we can use localVue so that testing doesn't pollute the config.
+// (unless we create a clearConfig() method to reset it)
+const CONFIG = {}
+
+// Method to get a deep clone (immutable) copy of the defaults
+const getDefaults = () => JSON.parse(JSON.stringify(DEFAULTS))
+
+// Method to set the config.
+// Merges in only known top-level and sub-level keys.
+//
+//   Vue.use(BootstrapVue, config)
+// or
+//   BootstrapVue.config(config)
+//   Vue.use(BootstrapVue)
+//
+// Breakpoint definition may need to be moved out of the config object
+// and set globally before bootstrapVue is loaded
+
+/* istanbul ignore next: just for now to prevent red X on codecov until we can test this */
+const setConfig = (opts = {}) => {
+  if (isObject(opts)) {
+    keys(opts).forEach(component => {
+      if (opts.hasOwnProperty(component) && DEFAULTS.hasOwnProperty(component)) {
+        if (component === 'breakpoints' && isArray(opts.breakpoints)) {
+          // special case for breakpoints
+          CONFIG.breakpoints = JSON.parse(JSON.stringify(opts.breakpoints))
+        } else if (isObject(opts[component])) {
+          keys(opts[component]).forEach(prop => {
+            if (opts[component].hasOwnProperty(prop) && DEFAULTS[component].hasOwnProperty(prop)) {
+              // If we pre-populate the config with the defaults, we can skip this line
+              CONFIG[component] = CONFIG[component] || {}
+              CONFIG[component][prop] = opts[component][prop]
+            }
+          })
+        }
+      }
+    })
+  }
+}
+
+// Method to grab a config value based on a dotted/array notation key.
+// Returns a deep clone (immutable) copy
+const getConfigParam = key => {
+  // First we try the user config, and if key not found we fall back to default value.
+  // NOTE: If we deep clone DEFAULTS into config, then we can skip the fallback for get
+  return JSON.parse(JSON.stringify(get(CONFIG, key, get(getDefaults(), key))))
+}
+
+// Method to grab a config value for a particular component.
+// Returns a deep clone (immutable) copy
+const getConfigComponent = (cmpName, key = null) => {
```

Comment:
I would suggest to rename this to `getComponentConfig`.

Reply:
changed